tests/test_unit/test_utils.py

```diff
@@ -0,0 +1,51 @@
+import glob
+import tempfile
+
+import pytest
+import yaml
+
+from wazp.utils import df_from_metadata_yaml_files
+
+
+def get_sample_project_metadata_fields() -> dict:
+    """Get the metadata dictionary from the sample project for testing."""
+    with open("sample_project/metadata_fields.yaml") as fi:
+        metadata_fields = yaml.safe_load(fi)
+    return metadata_fields
+
+
+def test_df_from_metadata_yaml_sample_project_metadata() -> None:
+    """Normal operation: test we can read the sample project metadata."""
+    fields = get_sample_project_metadata_fields()
+    testme = df_from_metadata_yaml_files("sample_project/videos", fields)
+
+    expected = set(fields)
+    actual = set(testme.columns)
+    diff = expected.symmetric_difference(actual)
```

Comment:
should we compute diff only if the don't match?

Reply:
For efficiency? I was wondering about that too but I couldn't come up with a snappy way to do it. [According to the python wiki `symmetric_difference` is only O(n)](https://wiki.python.org/moin/TimeComplexity) and we're not going crazy with metadata, right? So n ~ 30.